pkg/scheduler/scheduler.go

```diff
@@ -582,7 +589,12 @@ func (e entryOrdering) Less(i, j int) bool {
 		return !aBorrows
 	}
 
-	// 2. Higher priority first if not disabled.
+	// 2. Fair share, if enabled.
+	if e.enableFairSharing && a.dominantResourceShare != b.dominantResourceShare {
+		return a.dominantResourceShare < b.dominantResourceShare
+	}
+
+	// 3. Higher priority first if not disabled.
```

Comment:
IIUC here we prioritize scheduling based on fair share, but they are still not fully ignored as suggested in KEP goals: "When enforcing fair sharing, ignore workload priorities unless: (... point about super high prioirites... )"

They are used to break the tie, which will happen when multiple CQs are still have DRF <=0. So, there is still some incentive to set higher priority. Do you think this is acceptable, or should we fallback directly to time creation when Fair Sharing is used?

Reply:
For some organizations, it might still be useful to allow higher priority jobs to be admitted first.

It all depends on whether the organization has a centralized set of priorities, and somehow they are able to restrict their usage.

Reply:
> For some organizations, it might still be useful to allow higher priority jobs to be admitted first.

Right, certainly there is room for some configuration in the future. For now I'm just questioning if the tie-breaking by priority is desired when fair sharing is used. The KEP goals say: https://github.com/kubernetes-sigs/kueue/tree/main/keps/1714-fair-sharing#goals
```
When enforcing fair sharing, ignore workload priorities unless:

** The workload's priority is above admin-defined high priority. Super high priority workloads overrule fair sharing and are treated according to KEP [#1337](https://github.com/kubernetes-sigs/kueue/tree/main/keps/1337-preempt-within-cohort-while-borrowing).
```
This suggests to me that the tie breaker is creation time (bypassing priority) completely. I'm fine either way, just checking if this is not an omission.

Reply:
The KEP missed the point that whether to use priority as a tie-breaker is guarded by a feature gate already.

I'm not sure what to do with the priority long term, but for now, it makes sense to keep it guarded.

Reply:
Sgtm, so one can use `PrioritySortingWithinCohort` independently,

Reply:
> The KEP missed the point that whether to use priority as a tie-breaker is guarded by a feature gate already.
> 
> I'm not sure what to do with the priority long term, but for now, it makes sense to keep it guarded.

+1